devito/interfaces.py

```diff
@@ -72,7 +73,20 @@ def __new__(cls, *args, **kwargs):
     def __init__(self):
         """Initialise from a cached instance by shallow copying __dict__."""
         original = _SymbolCache[self.__class__]
-        self.__dict__ = original.__dict__.copy()
+        self.__dict__ = original().__dict__.copy()
+
+    def clean(self):
+        """Cleans up a :class:`SymbolicData` object by removing
+        it from the symbolic cache.
+        """
+        if self._cached():
+            self._clean()
```

Comment:
No, so to recap, `Operator.__del__` invokes `Operator.clear`, so that we auto-clear the sympy cache at the end of each test/inversion loop, right? Can we confirm that works?

The `clear_params` then might still be useful, just make sure you call the removal explicitly, ie. `del param`.